### PR TITLE
Reduce Math object binary size.

### PR DESCRIPTION
**Pull request by zherczeg**
Roughly one KByte saving: 141024 -> 140256

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 1]
```diff
@@ -49,596 +84,289 @@
...
- * @return ecma value
- *         Returned value must be freed with ecma_free_value.
- */
-static ecma_value_t
-ecma_builtin_math_object_cos (ecma_value_t this_arg, /**< 'this' argument */
-                              ecma_value_t arg) /**< routine's argument */
-{
-  JERRY_UNUSED (this_arg);
-  ecma_value_t ret_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY);
+     /* both numbers are zeroes */
```
Comment (tilmannOSG) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: Comment should start with uppercase letter and end with full stop. Though honestly I think you can just drop the comment :)

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 2]
```diff
@@ -49,596 +84,289 @@
...
- *          ECMA-262 v5, 15.8.2.8
+ *          ECMA-262 v5, 15.8.2.14
  *
  * @return ecma value
  *         Returned value must be freed with ecma_free_value.
  */
 static ecma_value_t
-ecma_builtin_math_object_exp (ecma_value_t this_arg, /**< 'this' argument */
-                              ecma_value_t arg) /**< routine's argument */
+ecma_builtin_math_object_random () /**< 'this' argument */
```
Comment (tilmannOSG) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: Should have a `void` between the parentheses and the comment removed. Or alternatively a `this` parameter added :)

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 3]
```diff
@@ -49,596 +84,289 @@
...
-} /* ecma_builtin_math_object_exp */
+  return ecma_make_number_value (((ecma_number_t) rand ()) / rand_max * rand_max_min_1 / rand_max);
+} /* ecma_builtin_math_object_random */
 
 /**
- * The Math object's 'floor' routine
- *
- * See also:
- *          ECMA-262 v5, 15.8.2.9
+ * Dispatcher of the built-in's routines
```
Comment (tilmannOSG) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: of -> for  
Please add a full stop as well.

**Comment by zherczeg** [comment]
Fixed all requests.

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 4]
```diff
@@ -49,596 +84,288 @@
...
-
-  ret_value = ecma_make_number_value (DOUBLE_TO_ECMA_NUMBER_T (asin (arg_num)));
-
-  ECMA_OP_TO_NUMBER_FINALIZE (arg_num);
-  return ret_value;
-} /* ecma_builtin_math_object_asin */
-
-/**
- * The Math object's 'atan' routine
+ * The Math object's 'max' 'min' routines
```
Comment (tilmannOSG) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: Full stop at the line end please.

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 5]
```diff
@@ -49,596 +84,288 @@
...
 
-  ECMA_OP_TO_NUMBER_FINALIZE (arg_num);
-  return ret_value;
-} /* ecma_builtin_math_object_cos */
+  return ecma_make_number_value (result_num);
+} /* ecma_builtin_math_object_max_min */
 
 /**
- * The Math object's 'exp' routine
+ * The Math object's 'random' routine
```
Comment (tilmannOSG) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: Full stop at the end of the sentence please.

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 6]
```diff
@@ -49,596 +84,288 @@
...
- *          ECMA-262 v5, 15.8.2.8
+ *          ECMA-262 v5, 15.8.2.14
  *
  * @return ecma value
  *         Returned value must be freed with ecma_free_value.
  */
 static ecma_value_t
-ecma_builtin_math_object_exp (ecma_value_t this_arg, /**< 'this' argument */
-                              ecma_value_t arg) /**< routine's argument */
+ecma_builtin_math_object_random ()
```
Comment (tilmannOSG) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: Please replace `ecma_builtin_math_object_random()` with `ecma_builtin_math_object_random(void)`. In C there's quite a difference between the two, right now you can call `ecma_builtin_math_object_random()` with an arbitrary number of arguments and the compiler won't complain :)

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 7]
```diff
@@ -49,596 +84,288 @@
...
-} /* ecma_builtin_math_object_exp */
+  return ecma_make_number_value (((ecma_number_t) rand ()) / rand_max * rand_max_min_1 / rand_max);
+} /* ecma_builtin_math_object_random */
 
 /**
- * The Math object's 'floor' routine
- *
- * See also:
- *          ECMA-262 v5, 15.8.2.9
+ * Dispatcher for the built-in's routines
```
Comment (tilmannOSG) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c]: Full stop at the end of the sentence please.

**Comment by zherczeg** [comment]
Thank you for the review. Issues were fixed.

**Pull request merged**